torsion_target/make_torsion_target.py

```diff
@@ -199,25 +200,40 @@ def make_torsiondrive_target(dataset_name, torsiondrive_data, test_ff=None):
                 target_mol.qm_grads.append(td_data['final_gradients'][grid_id])
             target_mol.write('scan.xyz')
             target_mol.write('qdata.txt')
-            # pick metadata to write into the metadata.json file
-            metadata = copy.deepcopy(td_data['keywords'])
-            metadata['dataset_name'] = dataset_name
-            metadata['entry_label'] = entry_index
-            metadata['canonical_smiles'] = td_data['attributes'].get('canonical_smiles', 'unknown')
-            metadata['torsion_grid_ids'] = sorted_grid_ids
-            # find SMIRKs for torsion being scaned if test_ff is provided
-            if test_ff:
-                metadata['smirks'] = []
-                metadata['smirks_ids'] = []
-                for torsion_indices in td_data['keywords']['dihedrals']:
-                    param = molecule_labels['ProperTorsions'][tuple(torsion_indices)]
-                    metadata['smirks'].append(param.smirks)
-                    metadata['smirks_ids'].append(param.id)
-            with open('metadata.json', 'w') as jsonfile:
-                json.dump(metadata, jsonfile, indent=2)
-            # finish this target
-            target_names.append(target_name)
-            os.chdir('..')
+            # check if the torsion scan contains one or more conformers forming strong internal H bonds
+            no_hbonds, hbonds = screening_Hbond(mol2_fnm='input.mol2', scan_fnm ='scan.xyz')
+            if no_hbonds != True:
```

Comment:
You might have noticed that this block is duplicated from the above (line 170). Let's actually move the `check_hbonds` feature between line 169 and 170, and we can set `success = False` when hydrogen bond is found.